clients/src/main/java/org/apache/kafka/common/security/scram/internals/ScramServerCallbackHandler.java

```diff
@@ -47,6 +49,28 @@ public ScramServerCallbackHandler(CredentialCache.Cache<ScramCredential> credent
     @Override
     public void configure(Map<String, ?> configs, String mechanism, List<AppConfigurationEntry> jaasConfigEntries) {
         this.saslMechanism = mechanism;
+
+        for (AppConfigurationEntry cfg: jaasConfigEntries) {
+            Map<String, ?> options = cfg.getOptions();
+            ScramFormatter formatter;
+
+            ScramMechanism scramMechanism = ScramMechanism.forMechanismName(mechanism);
+            try {
+                formatter = new ScramFormatter(scramMechanism);
+            } catch (NoSuchAlgorithmException e) {
+                return;
+            }
+
+            for (Map.Entry<String, ?> entry : options.entrySet()) {
+                if (entry.getKey().startsWith(JAAS_USER_PREFIX)) {
+                    String username = entry.getKey().replaceFirst(JAAS_USER_PREFIX, "");
+                    String password = (String) entry.getValue();
+
+                    credentialCache.put(username, formatter.generateCredential(password, 4096));
```

Comment:
I'm just not sure about the hard-coded number of iterations here. The value is fine, as it is the minIteration I see from [here](https://github.com/apache/kafka/blob/95947d2f581cce85c25ef9eb3501dc13d09ae05a/clients/src/main/java/org/apache/kafka/common/security/scram/internals/ScramMechanism.java#L33), although we could probably also use 8192 to be safer. 

I'm just wondering whether we should allow for the customization here, like allowing another config options called "iteration" or fall back to the minValue as default when not provided.